src/main/java/org/zalando/nakadi/domain/SubscriptionEventTypeStats.java

```diff
@@ -27,38 +26,13 @@ public String getEventType() {
     }
 
     @Immutable
+    @Getter
+    @AllArgsConstructor
     public static class Partition {
         private final String partition;
         private final String state;
         @JsonInclude(JsonInclude.Include.NON_NULL)
         private final Long unconsumedEvents;
         private final String streamId;
-
-        public Partition(final String partition,
-                         final String state,
-                         @Nullable final Long unconsumedEvents,
-                         final String streamId) {
-            this.partition = partition;
-            this.state = state;
-            this.unconsumedEvents = unconsumedEvents;
-            this.streamId = streamId;
-        }
-
-        public String getPartition() {
-            return partition;
-        }
-
-        public String getState() {
-            return state;
-        }
-
-        @Nullable
```

Comment:
@youssefmamdouh we try to follow a rule for null handling in this project (I guess it's not written anywhere, unfortunately).

Every method that could return a `null` is marked with `@Nullable` and the code that uses this method should be aware of that and treat it properly when necessary. Otherwise you can just consider that all other methods never return null.

This way we have been able to avoid null pointer problems so far.

Replacing all this code by a `getter/setter` annotation would be fine if we did not loose the `nullable` annotation. Is it possible to fix this?